### PR TITLE
[SPARK-36345][SPARK-36367][INFRA][PYTHON] Disable tests failed by the incompatible behavior of pandas 1.3

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Disable tests failed by the incompatible behavior of pandas 1.3.

### Why are the changes needed?

Pandas 1.3 has been released.
There are some behavior changes and we should follow it, but it's not ready yet.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Disabled some tests related to the behavior change.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Review by dongjoon-hyun — APPROVED** [review]
BTW, this is a blocker for 3.2?

**Comment by ueshin** [comment]
The other tests failed. I guess we should just revert #33595 for now.

**Comment by dongjoon-hyun** [comment]
Got it. Let me revert #33595 .

**Comment by ueshin** [comment]
@dongjoon-hyun 

> BTW, this is a blocker for 3.2?

That's a good question.

IMO, it's okay to say pandas API on Spark 3.2 follows the pandas 1.2 behavior.
but if we want to follow the pandas 1.3 behavior, this or SPARK-36367 is a blocker.

cc @HyukjinKwon

**Comment by HyukjinKwon** [comment]
I think it depends on how serious the breakage is (seems not super serious?). Would be great to fix it in at Spark 3.2 in any event  though.

**Comment by ueshin** [comment]
Let me use this PR to check how many tests fail.

File: python/pyspark/pandas/groupby.py [hunk 1]
```diff
@@ -1171,9 +1173,9 @@ def apply(self, func: Callable, *args: Any, **kwargs: Any) -> Union[DataFrame, S
 
         if is_series_groupby:
             name = psdf.columns[-1]
-            pandas_apply = SelectionMixin._builtin_table.get(func, func)
```
Comment (ueshin) [inline python/pyspark/pandas/groupby.py:1174]: @HyukjinKwon Do you remember what this was for?

Comment (HyukjinKwon) [reply python/pyspark/pandas/groupby.py:1174]: No .. the change seems fine

Comment (HyukjinKwon) [reply python/pyspark/pandas/groupby.py:1174]: Maybe it's just for better performance by switching to NumPy's?

Comment (ueshin) [reply python/pyspark/pandas/groupby.py:1174]: I see. Let's keep it as is for now.

**Comment by HyukjinKwon** [comment]
Had an offline discussion. Maybe I will just merge this into master branch only to minimise the impact on spark 3.2 preparation.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**